packages/components/transfer/src/composables/use-check.ts

```diff
@@ -1,54 +1,39 @@
-// @ts-nocheck
-import { computed, getCurrentInstance, watch } from 'vue'
-
-import type { ExtractPropTypes } from 'vue'
-import type { Key, TransferPanelState } from './transfer'
-
-export const CHECKED_CHANGE_EVENT = 'checked-change'
-
-export const useCheckProps = {
-  data: {
-    type: Array,
-    default() {
-      return []
-    },
-  },
-  optionRender: Function,
-  placeholder: String,
-  title: String,
-  filterable: Boolean,
-  format: Object,
-  filterMethod: Function,
-  defaultChecked: Array,
-  props: Object,
-}
+import { computed, watch } from 'vue'
+import { isFunction } from '@element-plus/utils'
+import { CHECKED_CHANGE_EVENT } from '../transfer-panel'
+import { usePropsAlias } from './use-props-alias'
+
+import type { SetupContext } from 'vue'
+import type { CheckboxValueType } from '@element-plus/components/checkbox'
+import type { TransferKey } from '../transfer'
+import type {
+  TransferPanelEmits,
+  TransferPanelProps,
+  TransferPanelState,
+} from '../transfer-panel'
 
 export const useCheck = (
-  props: ExtractPropTypes<typeof useCheckProps>,
-  panelState: TransferPanelState
+  props: TransferPanelProps,
+  panelState: TransferPanelState,
+  emit: SetupContext<TransferPanelEmits>['emit']
 ) => {
-  const { emit } = getCurrentInstance()
-
-  const labelProp = computed(() => props.props.label || 'label')
-
-  const keyProp = computed(() => props.props.key || 'key')
-
-  const disabledProp = computed(() => props.props.disabled || 'disabled')
+  const propsAlias = usePropsAlias(computed(() => props.props))
 
   const filteredData = computed(() => {
     return props.data.filter((item) => {
-      if (typeof props.filterMethod === 'function') {
+      if (isFunction(props.filterMethod)) {
         return props.filterMethod(panelState.query, item)
       } else {
-        const label = item[labelProp.value] || item[keyProp.value].toString()
+        const label =
+          item[propsAlias.value.label] || item[propsAlias.value.key].toString()
```

Comment:
```suggestion
          item[propsAlias.value.label] || item[propsAlias.value.key].toString()
          item[propsAlias.value.label] || String(item[propsAlias.value.key])
```

String() function is recommended

Reply:
Doing this also doesn't solve the bug you mentioned above. 🤔

PTAL https://github.com/element-plus/element-plus/pull/8343#discussion_r905640568

Reply:
Yes, this does prevent the component from crashing, but it also introduces new issues:

https://user-images.githubusercontent.com/23100055/175532536-e46a6a14-cca5-4520-9b16-92540e5576d3.mov

Reply:
All the same values ​​will of course all be selected, you can try inserting some normal data.

Reply:
Got it, thank you!